### PR TITLE
Run tests in parallel to speed up CI performance

### PR DESCRIPTION
**Pull request by mikenachbaur-okta**

### CODE REVIEW

**Review by rajdeepnanua-okta — APPROVED** [review]

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -39,52 +41,77 @@ jobs:
...
-          -destination "platform=iOS Simulator,OS=latest,name=iPhone 12" | xcpretty
+          -destination "$iOS_DESTINATION" | xcpretty
     - name: Upload Test Results
       uses: actions/upload-artifact@v3
       with:
-        path: ~/TestResults/*.xcresult
-        name: XcodeBuild-Results
-        if-no-files-found: warn
+        path: ../iOS-Build.tgz
+        name: iOS-Build
```
Comment (rajdeepnanua-okta) [inline .github/workflows/tests.yaml:113]: This is no longer uploading test results

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -95,40 +122,38 @@ jobs:
...
     - uses: ./.github/actions/setup-secrets
       with:
         okta_plist: "${{ secrets.TEST_OKTA_PLIST }}"
         test_configuration: "${{ secrets.TEST_CONFIGURATION }}"
     - name: Test Web Sign In
       run: |
         xcodebuild \
-            -derivedDataPath ~/Build/DerivedData \
-            -clonedSourcePackagesDirPath ~/Build/ClonedSources \
-            -resultBundlePath ~/TestResults/WebSignIn-iOS.xcresult \
```
Comment (rajdeepnanua-okta) [inline .github/workflows/tests.yaml:107]: Why was ```-resultBundlePath ~/TestResults/WebSignIn-iOS.xcresult``` removed?

**Review by rajdeepnanua-okta — COMMENTED** [review]
Generally looks good. I had 2 questions.

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (mikenachbaur-okta) [reply .github/workflows/tests.yaml:113]: Yes, the test result post-processing was error prone and added several minutes to the build time. I didn't feel like it was adding that much value, so I'm taking it out.

File: .github/workflows/tests.yaml (hunk 2 above)
Comment (mikenachbaur-okta) [reply .github/workflows/tests.yaml:107]: Exporting the result bundle is only necessary if you want to post-process the test results, which includes screen shots of every step of the UI tests.

**Pull request merged**